core/src/main/scala/com/databricks/labs/remorph/graph/TableDependencyGraph.scala

```diff
@@ -0,0 +1,139 @@
+package com.databricks.labs.remorph.graph
+
+import com.databricks.labs.remorph.discovery.{ExecutedQuery, QueryHistory, TableDefinition}
+import com.databricks.labs.remorph.parsers.PlanParser
+import com.databricks.labs.remorph.transpilers.{Result, SourceCode}
+import com.typesafe.scalalogging.LazyLogging
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+
+protected case class Node(tableDefinition: TableDefinition, metadata: Map[String, Set[String]])
+protected case class Edge(from: TableDefinition, to: TableDefinition, metadata: Map[String, String])
+
+class TableDependencyGraph(parser: PlanParser[_]) extends DependencyGraph with LazyLogging {
```

Comment:
```suggestion
class TableGraph(parser: PlanParser[_], tables: Set[TableDefinition]) extends DependencyGraph with LazyLogging {
```
shorter names are always better. also, i think it's better to depend on a precomputed set of table definitions, that we retrieve separately.

Reply:
isn't it better to pass the `tabledefinition` to the public method for this class. That way we create instance of one object per sourcesystem.